src/skadi/sample.cc

```diff
@@ -72,42 +68,105 @@ namespace {
   }
 
   uint64_t file_size(const std::string& file_name) {
+    //TODO: detect gzip and actually validate the uncompressed size?
     struct stat s;
     int rc = stat(file_name.c_str(), &s);
     return rc == 0 ? s.st_size : -1;
   }
 
+  void gunzip(const valhalla::midgard::mem_map<char>& in, int16_t* tile) {
```

Comment:
both forms of decompression work with bytes that are already `mmap`ped this increased performance by 15% and made the difference between `lz4` and `raw` basically 0 for the nominal use cases